File: movie-api/src/main/java/com/example/app/movie/presentation/dto/request/MovieSearchRequest.java
package com.example.app.movie.presentation.dto.request;

import com.example.app.common.annotation.ValidEnum;
import com.example.app.movie.dto.SearchMovieCommand;
import com.example.app.movie.type.MovieGenre;
import org.hibernate.validator.constraints.Length;

import static java.util.Objects.nonNull;

public record MovieSearchRequest(
        @Length(max = 255, message = "제목은 255자 이하로 검색해주세요")
        String title,
        @ValidEnum(enumClass = MovieGenre.class)
        String genre
){
        public SearchMovieCommand toMovieSearchCommand() {
                return SearchMovieCommand.builder()
                        .title(this.title)
                        .genre(nonNull(this.genre) ? MovieGenre.valueOf(this.genre) : null)
                        .build();
        }
}

File: movie/api/src/main/kotlin/com/example/movie/request/MovieSearchRequest.kt
package com.example.movie.request

import com.example.movie.domain.screening.model.ScreeningStatus
import jakarta.validation.Valid
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.Size

data class MovieSearchRequest(
    @field:Valid
    val status: ScreeningStatus,

    @field:Size(max = 100, message = "제목은 100자를 초과할 수 없습니다")
    val title: String?,

    @field:Min(value = 1, message = "장르 ID는 1 이상이어야 합니다")
    val genreId: Long?
)